addons/google_account/controllers/main.py

```diff
@@ -20,7 +20,9 @@ def oauth2callback(self, **kw):
 
         with registry(dbname).cursor() as cr:
             if kw.get('code'):
-                request.env(cr, request.session.uid)['google.%s' % service].set_all_tokens(kw['code'])
+                access_token, refresh_token, ttl = request.env['google.service']._get_google_tokens(kw['code'], service)
+                # LUL TODO only defined in google_calendar
```

Comment:
``_get_google_tokens`` is only called once. Maybe you could override the method on google_calendar right ?